apis/python/tests/test_dense_nd_array.py

```diff
@@ -60,6 +60,28 @@ def test_dense_nd_array_create_ok(
         assert isinstance(A._handle._handle, soma.pytiledbsoma.SOMADenseNDArray)
 
 
+def test_dense_nd_array_reopen(tmp_path):
```

Comment:
suggest adding tests that confirm error handling works and the right exception is raised.  E.g., `reopen("bad-mode")`, or reopening a closed array/group.

Reply:
another test that would be useful:  confirm that the reopen bumps timestamp to latest.